problem_builder/public/js/plot.js

```diff
@@ -1,5 +1,23 @@
 function PlotBlock(runtime, element) {
 
+    // jQuery helpers
+
+    jQuery.fn.isEmpty = function() {
+        return !$.trim($(this).html());
+    };
+
+    jQuery.fn.isHidden = function() {
+        // Don't use jQuery :hidden selector here;
+        // this is necessary to ensure that result is independent of parent visibility
+        return $(this).css('display') === 'none';
```

Comment:
@Kelketek This is to ensure all edge cases can be handled properly while keeping toggling behavior of information for individual overlays transparent. By the latter I mean that we can keep the logic for showing overlay info simpler if we make sure that the display status of overlay *info* always corresponds to the display status of the corresponding overlay. If an overlay is on, associated info will be "visible" (as in, `display: block`); if it's off, associated info will be hidden (as in, `display: none`). This requirement alone does preclude using `:hidden`/`:visible`, but there are a couple more aspects to consider:

1. The parent element that groups overlay info includes a header ("Plot info") that we want to show when info about one or more overlays is visible.
2. Some overlays have neither a description nor a citation. In that case, we don't really have any info to show (plot label doesn't count since it's already used for the button that toggles the overlay).
3. We would like to avoid showing the "Plot info" header if there is no info to display; i.e., if all overlays are either off or don't have any info associated with them.

So if we start out with all overlays off and then turn on a single overlay that doesn't have a description or a citation, we don't want to show the parent element. As a result, the parent element will have `display: none`, while the child will have `display: block`. At this point, we can't reliably use `.is(":visible")` on the child anymore to determine whether the child should be shown or not because if the parent is visible, it considers the child to be visible as well. This is why I'm not using `:hidden`/`:visible` here.

---

Writing this out now I realize that instead of toggling visibility of the parent, it would probably also be possible to keep the parent visible at all times and just toggle visibility of the "Plot info" header. In that case we could use `:hidden`/`:visible` here. However, I'm not sure if that would yield the cleanest solution in terms of what we do and don't show on the page at any given time: If we show the parent even if overlay info for all active overlays is "empty", the parent will end up taking some space on the page because the children are not really empty (they appear to be, but they do contain some whitespace). So yeah, overall I think I slightly favor the current approach, which could be summarized as "If we only have whitespace to show, we'd rather not show anything at all".

Let me know what you think; I'll hold of on merging for now just in case.